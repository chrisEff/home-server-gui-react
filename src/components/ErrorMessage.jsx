'use strict'

import React from 'react'
import {StyleSheet, css} from 'aphrodite'

const ErrorMessage = ({message}) =>
	<div className={css([styles.errorMessage])}>{message}</div>

const styles = StyleSheet.create({
	errorMessage: {
		color: 'red',
	},
})

export default ErrorMessage
