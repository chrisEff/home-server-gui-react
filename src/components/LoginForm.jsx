'use strict'

const React = require('react')

class LoginForm extends React.Component {

	render () {
		return (
			<div style={{textAlign: 'center'}}>
				<br/><br/>

				API URL:<br/>
				<input type='text' ref='apiUrl' style={{width: 250}}/><br/><br/><br/>

				API Key:<br/>
				<input type='text' ref='apiKey' style={{width: 250}}/><br/><br/><br/>

				<button onClick={ () => this.props.onLogin(this.refs.apiUrl.value, this.refs.apiKey.value) }>OK</button>
			</div>
		)
	}

}

module.exports = LoginForm
