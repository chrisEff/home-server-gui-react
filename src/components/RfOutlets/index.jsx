'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'
import {css} from 'aphrodite'

import RfOutlet from './RfOutlet'

import {loadOutlets} from '@/actions/outlets'
import theme from '@/theme'

class RfOutlets extends React.Component {

	static propTypes = {
		title: PropTypes.string.isRequired,
		outlets: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number,
			})
		),
	}

	async componentDidMount () {
		this.props.onLoad()
	}

	render () {
		return !this.props.outlets ? <div/> : (
			<div className={css([theme.styles.section])}>
				<h2>{this.props.title}</h2>
				<div className={css([theme.styles.sectionBody])}>
					{this.props.outlets.map(outlet =>
						<RfOutlet key={outlet.id} id={outlet.id}/>
					)}
				</div>
			</div>
		)
	}
}

const mapStateToProps = state => ({
	outlets: state.outlets,
})

const mapDispatchToProps = dispatch => ({
	onLoad: () => dispatch(loadOutlets()),
})

export default connect(mapStateToProps, mapDispatchToProps)(RfOutlets)
