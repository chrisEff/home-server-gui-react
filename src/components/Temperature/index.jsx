'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'

import TemperatureSensor from './TemperatureSensor'

import {loadTempSensors} from '../../actions/temperature'

class Temperature extends React.Component {

	static propTypes = {
		title: PropTypes.string,
		tempSensors: PropTypes.arrayOf(
			PropTypes.object
		),
	}

	async componentDidMount () {
		this.props.onLoad()
	}

	render () {
		return !this.props.tempSensors ? <div/> : (
			<div id='temperature'>
				<h2>{this.props.title}</h2>
				{this.props.tempSensors.map(sensor =>
					<TemperatureSensor key={sensor.id} name={sensor.name} value={sensor.celsiusValue} history={sensor.history}/>
				)}
			</div>
		)
	}
}

const mapStateToProps = state => ({
	tempSensors: state.tempSensors,
})

const mapDispatchToProps = dispatch => ({
	onLoad: () => dispatch(loadTempSensors()),
})

export default connect(mapStateToProps, mapDispatchToProps)(Temperature)
