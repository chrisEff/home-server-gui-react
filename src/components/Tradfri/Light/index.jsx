'use strict'

const homeServerApi = require('../../../homeServerApi')

const React = require('react')
const PropTypes = require('prop-types')

const LightTitle = require('./LightTitle')
const BrightnessSlider = require('./BrightnessSlider')
const ColorSwitcher = require('./ColorSwitcher')
const LightSwitch = require('./LightSwitch')

class Light extends React.Component {

	static propTypes = {
		bulb: PropTypes.object.isRequired,
	}

	constructor (props) {
		super(props)
		this.state = {bulb: props.bulb}
	}

	render () {
		const bulb = this.state.bulb
		const className = bulb.state ? 'light on' : 'light'
		return (
			<div className={className + ' ' + bulb.color}>
				<LightTitle title={bulb.name} onChange={this.updateName} />

				<LightSwitch state={bulb.state} onchange={this.updateState}/>
				<BrightnessSlider brightness={bulb.brightness} onchange={this.updateBrightness}/>
				<ColorSwitcher bulbType={bulb.bulbType} color={bulb.color} onchange={this.updateColor}/>
			</div>
		)
	}

	updateName = async (name) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/name/${name}`)

		bulb.name = name
		this.setState({bulb})
	}

	updateState = async (state) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/state/${state}`)

		bulb.state = state
		this.setState({bulb})
	}

	updateBrightness = async (brightness) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/brightness/${brightness}`)

		bulb.brightness = brightness
		this.setState({bulb})
	}

	updateColor = async (color) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/color/${color}`)

		bulb.color = color
		this.setState({bulb})
	}
}

module.exports = Light
