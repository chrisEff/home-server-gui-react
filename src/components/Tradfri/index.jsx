'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'

import LightGroup from './LightGroup'
import {loadTradfriDevices, loadTradfriGroups} from '../../actions'

class Tradfri extends React.Component {

	static propTypes = {
		title: PropTypes.string,
	}

	async componentDidMount () {
		this.props.onLoad()
	}

	render () {
		return (
			<div id='tradfri'>
				<h2>{this.props.title}</h2>
				{this.props.tradfriGroups.map(group => <LightGroup key={group.name} id={group.id} name={group.name} deviceIds={group.deviceIds}/>)}
			</div>
		)
	}
}

const mapStateToProps = state => ({
	tradfriGroups: state.tradfriGroups,
	tradfriDevices: state.tradfriDevices,
})

const mapDispatchToProps = dispatch => ({
	onLoad: () => {
		dispatch(loadTradfriGroups())
		dispatch(loadTradfriDevices())
	},
})

export default connect(mapStateToProps, mapDispatchToProps)(Tradfri)
export {Tradfri}
