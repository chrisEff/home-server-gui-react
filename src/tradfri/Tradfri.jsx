'use strict'

const config = require('../../config')
const LightGroup = require('./LightGroup')
const ErrorMessage = require('../ErrorMessage')

class Tradfri extends React.Component {
	constructor (props) {
		super(props)
		this.state = {
			groups: [],
			errorMsg: null,
		}
	}

	async componentDidMount () {
		try {
			const groups = await (await fetch(`${config.api.protocol}://${config.api.host}:${config.api.port}/tradfri/group?key=${config.api.key}`)).json()
			const devices = await (await fetch(`${config.api.protocol}://${config.api.host}:${config.api.port}/tradfri/device?key=${config.api.key}`)).json()

			groups.forEach(group => {
				group.devices = []
				group.deviceIds.forEach(deviceId => {
					const device = devices.find(device => device.id === deviceId)
					group.devices.push(device)
				})
				group.devices.sort((a, b) => {
					return a.name > b.name ? 1 : b.name < a.name ? -1 : 0;
				})
			})
			groups.sort((a, b) => {
				return a.name > b.name ? 1 : b.name < a.name ? -1 : 0;
			})

			this.setState({
				groups
			})
		} catch (e) {
			this.setState({errorMsg: e.message})
		}
	}

	render () {
		return (
			<div>
				<h2>Licht</h2>
				{this.state.errorMsg && <ErrorMessage message={this.state.errorMsg}/>}
				{this.state.groups.map(group => <LightGroup key={group.name} group={group}/>)}
			</div>
		)
	}
}

module.exports = Tradfri
