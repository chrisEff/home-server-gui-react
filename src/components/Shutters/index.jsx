'use strict'

import homeServerApi from '../../homeServerApi'

import React from 'react'
import PropTypes from 'prop-types'

import Shutter from './Shutter'
import ErrorMessage from '../ErrorMessage'

class Shutters extends React.Component {

	static propTypes = {
		title: PropTypes.string,
	}

	constructor (props) {
		super(props)
		this.state = {
			shutters: [],
			errorMsg: null,
		}
	}

	async componentDidMount () {
		try {
			const response = await homeServerApi.get(`/shutters/shutter`)
			const shutters = Object.values(await response.json())
			this.setState({shutters})
		} catch (e) {
			console.log(e)
			this.setState({errorMsg: e.message})
		}
	}

	render () {
		return (
			<div id='shutters'>
				<h2>{this.props.title}</h2>
				{this.state.errorMsg && <ErrorMessage message={this.state.errorMsg}/>}
				{this.state.shutters.map(shutter => <Shutter key={shutter.name} shutter={shutter}/>)}
			</div>
		)
	}
}

export default Shutters
