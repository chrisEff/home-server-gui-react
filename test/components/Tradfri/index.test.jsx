'use strict'

import React from 'react'
import {Tradfri} from '@/components/Tradfri'

import {shallow} from 'enzyme'

describe('Tradfri', () => {
	it('should render correctly', () => {
		const wrapper = shallow(<Tradfri title='test' tradfriGroups={[{
			id: 131092,
			name: 'Office',
			deviceIds: [65603, 65537, 65602],
		}]} />)
		expect(wrapper).toMatchSnapshot()
	})
})
