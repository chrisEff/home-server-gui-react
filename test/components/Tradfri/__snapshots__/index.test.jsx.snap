// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tradfri should render correctly 1`] = `
<div
  id="tradfri"
>
  <h2>
    test
  </h2>
  <ConnectFunction
    deviceIds={
      Array [
        65603,
        65537,
        65602,
      ]
    }
    id={131092}
    key="Office"
    name="Office"
  />
</div>
`;
